node/src/components/chainspec_loader.rs

```diff
@@ -210,7 +221,11 @@ impl ChainspecLoader {
         let mut effects = if should_stop {
             Effects::new()
         } else {
-            async { Some(Event::Initialize) }.map_some(std::convert::identity)
+            effect_builder
+                .get_highest_block_from_storage()
```

Comment:
Would it be enough to load the highest block _header_, without body?

Reply:
Yup - good point.  Same thing for the deploy acceptor - I'll make that change too since it's more frequently called and will be even more beneficial than this one which is only called once in a session.

Reply:
Fixed in [148a037](https://github.com/casper-network/casper-node/pull/2619/commits/148a037bdc00338d76e60afd30ff1d4834a10b95).